### ISSUE TITLE
Low accuracy on the inference model as compared to trained model

### ISSUE THREAD

**Issue by kajal-singh31** [PaddlePaddle/PaddleOCR#2425]
While working on text recognition inference, we observed a huge drop in accuracy of inference model while that for pre-trained model was comparatively better. 
We noticed our text recognition pretrained model (which was finetuned with custom dataset) had 96.5% accuracy while that for same exported inference model had 90.5% accuracy.

**Comment by tink2123** [comment]
Please provide the command line for infer prediction.

**Comment by kajal-singh31** [comment]
python3 tools/infer_rec.py -c configs/rec/rec_mv3_none_bilstm_ctc_finetune.yml -o Global.pretrained_model=./output/rec_2/best_accuracy Global.load_static_weights=false Global.infer_img=./train_data/data/
[2021/04/12 05:02:32] root INFO: Architecture : 
[2021/04/12 05:02:32] root INFO:     Backbone : 
[2021/04/12 05:02:32] root INFO:         model_name : large
[2021/04/12 05:02:32] root INFO:         name : MobileNetV3
[2021/04/12 05:02:32] root INFO:         scale : 0.5
[2021/04/12 05:02:32] root INFO:     Head : 
[2021/04/12 05:02:32] root INFO:         fc_decay : 0
[2021/04/12 05:02:32] root INFO:         name : CTCHead
[2021/04/12 05:02:32] root INFO:     Neck : 
[2021/04/12 05:02:32] root INFO:         encoder_type : rnn
[2021/04/12 05:02:32] root INFO:         hidden_size : 96
[2021/04/12 05:02:32] root INFO:         name : SequenceEncoder
[2021/04/12 05:02:32] root INFO:     Transform : None
[2021/04/12 05:02:32] root INFO:     algorithm : CRNN
[2021/04/12 05:02:32] root INFO:     model_type : rec
[2021/04/12 05:02:32] root INFO: Eval : 
[2021/04/12 05:02:32] root INFO:     dataset : 
[2021/04/12 05:02:32] root INFO:         data_dir : ./train_data/or_mdb/
[2021/04/12 05:02:32] root INFO:         name : LMDBDataSet
[2021/04/12 05:02:32] root INFO:         transforms : 
[2021/04/12 05:02:32] root INFO:             DecodeImage : 
[2021/04/12 05:02:32] root INFO:                 channel_first : False
[2021/04/12 05:02:32] root INFO:                 img_mode : BGR
[2021/04/12 05:02:32] root INFO:             CTCLabelEncode : None
[2021/04/12 05:02:32] root INFO:             RecResizeImg : 
[2021/04/12 05:02:32] root INFO:                 image_shape : [3, 32, 100]
[2021/04/12 05:02:32] root INFO:             KeepKeys : 
[2021/04/12 05:02:32] root INFO:                 keep_keys : ['image', 'label', 'length']
[2021/04/12 05:02:32] root INFO:     loader : 
[2021/04/12 05:02:32] root INFO:         batch_size_per_card : 256
[2021/04/12 05:02:32] root INFO:         drop_last : False
[2021/04/12 05:02:32] root INFO:         num_workers : 4
[2021/04/12 05:02:32] root INFO:         shuffle : True
[2021/04/12 05:02:32] root INFO: Global : 
[2021/04/12 05:02:32] root INFO:     cal_metric_during_train : True
[2021/04/12 05:02:32] root INFO:     character_dict_path : None
[2021/04/12 05:02:32] root INFO:     character_type : EN_symbol
[2021/04/12 05:02:32] root INFO:     checkpoints : None
[2021/04/12 05:02:32] root INFO:     debug : False
[2021/04/12 05:02:32] root INFO:     distributed : False
[2021/04/12 05:02:32] root INFO:     epoch_num : 500
[2021/04/12 05:02:32] root INFO:     eval_batch_step : [0, 500]
[2021/04/12 05:02:32] root INFO:     infer_img : ./train_data/data/
[2021/04/12 05:02:32] root INFO:     infer_mode : False
[2021/04/12 05:02:32] root INFO:     load_static_weights : False
[2021/04/12 05:02:32] root INFO:     log_smooth_window : 20
[2021/04/12 05:02:32] root INFO:     max_text_length : 150
[2021/04/12 05:02:32] root INFO:     pretrained_model : ./output/y/best_accuracy
[2021/04/12 05:02:32] root INFO:     print_batch_step : 10
[2021/04/12 05:02:32] root INFO:     save_epoch_step : 100
[2021/04/12 05:02:32] root INFO:     save_inference_dir : None
[2021/04/12 05:02:32] root INFO:     save_model_dir : ./output/rec_trt_lr0005_evalOrc_checkLoss/
[2021/04/12 05:02:32] root INFO:     use_gpu : True
[2021/04/12 05:02:32] root INFO:     use_space_char : False
[2021/04/12 05:02:32] root INFO:     use_visualdl : False
[2021/04/12 05:02:32] root INFO: Loss : 
[2021/04/12 05:02:32] root INFO:     name : CTCLoss
[2021/04/12 05:02:32] root INFO: Metric : 
[2021/04/12 05:02:32] root INFO:     main_indicator : acc
[2021/04/12 05:02:32] root INFO:     name : RecMetric
[2021/04/12 05:02:32] root INFO: Optimizer : 
[2021/04/12 05:02:32] root INFO:     lr : 
[2021/04/12 05:02:32] root INFO:         learning_rate : 0.0005
[2021/04/12 05:02:32] root INFO:     name : Adadelta
[2021/04/12 05:02:32] root INFO:     regularizer : 
[2021/04/12 05:02:32] root INFO:         factor : 0
[2021/04/12 05:02:32] root INFO:         name : L2
[2021/04/12 05:02:32] root INFO:     rho : 0.95
[2021/04/12 05:02:32] root INFO: PostProcess : 
[2021/04/12 05:02:32] root INFO:     name : CTCLabelDecode
[2021/04/12 05:02:32] root INFO: Train : 
[2021/04/12 05:02:32] root INFO:     dataset : 
[2021/04/12 05:02:32] root INFO:         data_dir : ./train_data/x/
[2021/04/12 05:02:32] root INFO:         name : LMDBDataSet
[2021/04/12 05:02:32] root INFO:         transforms : 
[2021/04/12 05:02:32] root INFO:             DecodeImage : 
[2021/04/12 05:02:32] root INFO:                 channel_first : False
[2021/04/12 05:02:32] root INFO:                 img_mode : BGR
[2021/04/12 05:02:32] root INFO:             CTCLabelEncode : None
[2021/04/12 05:02:32] root INFO:             RecResizeImg : 
[2021/04/12 05:02:32] root INFO:                 image_shape : [3, 32, 100]
[2021/04/12 05:02:32] root INFO:             KeepKeys : 
[2021/04/12 05:02:32] root INFO:                 keep_keys : ['image', 'label', 'length']
[2021/04/12 05:02:32] root INFO:     loader : 
[2021/04/12 05:02:32] root INFO:         batch_size_per_card : 256
[2021/04/12 05:02:32] root INFO:         drop_last : True
[2021/04/12 05:02:32] root INFO:         num_workers : 8
[2021/04/12 05:02:32] root INFO:         shuffle : True
[2021/04/12 05:02:32] root INFO: train with paddle 2.0.0 and device CUDAPlace(0)
W0412 05:02:32.243779 28171 device_context.cc:362] Please NOTE: device: 0, GPU Compute Capability: 7.5, Driver API Version: 11.1, Runtime API Version: 10.2
W0412 05:02:32.249138 28171 device_context.cc:372] device: 0, cuDNN Version: 7.6.
[2021/04/12 05:02:35] root INFO: load pretrained model from ['./output/rec_CL/best_accuracy']
[2021/04/12 05:02:35] root INFO: 	 result: ('16', 0.99999696)
[2021/04/12 05:02:35] root INFO: 	 result: ('?', 0.9999951)
[2021/04/12 05:02:35] root INFO: 	 result: ('File', 0.9995617)
[2021/04/12 05:02:35] root INFO: 	 result: ('T', 0.99999666)
[2021/04/12 05:02:35] root INFO: 	 result: ('Close', 0.9999995)
[2021/04/12 05:02:35] root INFO: 	 result: ('1', 0.99999917)
[2021/04/12 05:02:35] root INFO: 	 result: ('$1,063.00', 0.999997)
[2021/04/12 05:02:35] root INFO: 	 result: ('Edit', 0.98141223)
[2021/04/12 05:02:35] root INFO: 	 result: ('Name"', 0.99984264)
[2021/04/12 05:02:35] root INFO: 	 result: ('Total', 0.9911337)
[2021/04/12 05:02:35] root INFO: 	 result: ('Pivo...', 0.9536444)
[2021/04/12 05:02:35] root INFO: 	 result: ('$200.00', 0.99999875)
[2021/04/12 05:02:35] root INFO: 	 result: ('Type', 0.9999894)
[2021/04/12 05:02:35] root INFO: 	 result: ('Legal', 0.9997263)
[2021/04/12 05:02:35] root INFO: 	 result: ('S4H', 0.99999976)
[2021/04/12 05:02:35] root INFO: 	 result: ('Ground', 0.99997663)
[2021/04/12 05:02:35] root INFO: 	 result: ('a', 0.9999999)
[2021/04/12 05:02:35] root INFO: 	 result: ('has', 1.0)
[2021/04/12 05:02:35] root INFO: 	 result: ('expenses', 1.0)
[2021/04/12 05:02:35] root INFO: 	 result: ('try', 0.99601966)
[2021/04/12 05:02:35] root INFO: 	 result: ('11/17/2010', 0.9999884)
[2021/04/12 05:02:35] root INFO: 	 result: ('Internat.', 0.979727)
[2021/04/12 05:02:35] root INFO: 	 result: ('enter', 0.99756795)

**Comment by JITESH11989** [comment]
any update on this

**Comment by kajalthakur31** [comment]
Look out for image_shape before and after exporting. If you are running infer_rec with 3,32,100; then after export also change image shape in utility file to 3,32,100. This nearly fixed the issue.

**Comment by rsingh2083** [comment]
> While working on text recognition inference, we observed a huge drop in accuracy of inference model while that for pre-trained model was comparatively better.
> We noticed our text recognition pretrained model (which was finetuned with custom dataset) had 96.5% accuracy while that for same exported inference model had 90.5% accuracy.

How to train on additional images. Which labelling software you used ?
